packages/web3-providers-http/src/index.js

```diff
@@ -34,18 +34,19 @@ var https = require('https');
 var HttpProvider = function HttpProvider(host, options) {
     options = options || {};
 
+    this.withCredentials = options.withCredentials || false;
+    this.timeout = options.timeout || 0;
+    this.headers = options.headers;
+    this.agent = options.agent;
+    this.connected = false;
+
     var keepAlive = (options.keepAlive === true || options.keepAlive !== false) ? true : false;
     this.host = host || 'http://localhost:8545';
-    if (this.host.substring(0,5) === "https") {
+    if (!this.agent && this.host.substring(0,5) === "https") {
         this.httpsAgent = new https.Agent({ keepAlive: keepAlive });
     } else {
         this.httpAgent = new http.Agent({ keepAlive: keepAlive });
     }
```

Comment:
@nivida Could you eli5 me why http and https are treated differently here depending on whether `options.agent` is defined or not?

Reply:
Oh, thanks! I've missed the else statement 👌